slushfile.js

```diff
@@ -233,6 +238,18 @@ function configRoxy() {
       '          <range-value-positions>false</range-value-positions>\n' +
       '        </range-element-index>\n');
 
+    // fix default app-role privileges to match rest-style applications
+    foo = foo.replace(/<privileges>[^]*?<\/privileges>/,
+      '<privileges>\n' +
+      '        <privilege>\n' +
+      '          <privilege-name>rest-reader</privilege-name>\n' +
+      '        </privilege>\n' +
+      '        <!-- remove the rest-writer privilege if read-only access is required -->\n' +
+      '        <privilege>\n' +
+      '          <privilege-name>rest-writer</privilege-name>\n' +
+      '        </privilege>\n' +
+      '      </privileges>\n');
+
```

Comment:
Using rest-reader/writer privs also requires making sure REST extensions and such get app-role permissions, or otherwise calls those REST extensions will fail. This means patching app_specific with something like:

```
  alias_method :original_deploy_rest, :deploy_rest
  def deploy_rest
    original_deploy_rest

    r = execute_query(%Q{
        xquery version "1.0-ml";

        for $uri in cts:uris()
        return (
            $uri,
            xdmp:document-set-permissions($uri, (
                xdmp:permission("#{@properties["ml.app-name"]}-role", "read"),
                xdmp:permission("#{@properties["ml.app-name"]}-role", "execute")
            ))
       )
      },
      { :db_name => @properties["ml.modules-db"] }
    )
    r.body = parse_json r.body
    logger.debug r.body
  end
```